pkgs/applications/version-management/deepgit/default.nix

```diff
@@ -0,0 +1,86 @@
+{ copyDesktopItems
+, fetchurl
+, glib
+, gnome
+, gtk3
+, jre
+, lib
+, makeDesktopItem
+, stdenv
+, wrapGAppsHook
+}:
+
+stdenv.mkDerivation rec {
+  pname = "deepgit";
+  version = "4.3";
+
+  src = fetchurl {
+    url = "https://www.syntevo.com/downloads/deepgit/deepgit-linux-${lib.replaceStrings [ "." ] [ "_" ] version}.tar.gz";
+    hash = "sha256-bA/EySZjuSDYaZplwHcpeP1VakcnG5K1hYTk7cSVbz0=";
+  };
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    gnome.adwaita-icon-theme
+    gtk3
+    jre
+  ];
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ glib gtk3 ]}
```

Comment:
Could we avoid using LD_LIBRARY_PATH here?

Reply:
I was largely mirroring what was done in `pkgs.smartgithg`, at @hab25's suggestion; I am happy to apply a patch to this, if you feel that this is messy or bad form, but I am unclear what the alternative would look like

Reply:
LD_LIBRARY_PATH is always a bit dirty because it leaks into programs executed and is usually a good source for impurity and other trouble.

Reply:
I follow the abstract consequences you are describing. but am still unclear how this would be resolved, open to patches